vault/vault_secret_client.go

```diff
@@ -0,0 +1,137 @@
+package vault
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/rs/zerolog"
+)
+
+type VaultClient struct {
+	address     string
+	authPath    string
+	role        string
+	jwtFilePath string
+	logger      *zerolog.Logger
+}
+
+const ROLE = "role"
+const JWT = "jwt"
+
+// return structure for Vault Client, based on configuration
+func NewVaultClient(conf map[interface{}]interface{}, logger *zerolog.Logger) VaultClient {
+	address := "http://vault.fybrik-system:8200"
+	authPath := "/v1/auth/kubernetes/login"
+	role := "fybrik"
+	jwtFilePath := "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	if conf != nil {
+		if addressConf, ok := conf["address"]; ok {
+			address = addressConf.(string)
+		}
+		if authPathConf, ok := conf["authPath"]; ok {
+			authPath = authPathConf.(string)
+		}
+		if roleConf, ok := conf["role"]; ok {
+			role = roleConf.(string)
+		}
+		if jwtFilePathConf, ok := conf["jwt_file_path"]; ok {
+			jwtFilePath = jwtFilePathConf.(string)
+		}
+	}
+	return VaultClient{address: address, authPath: authPath, role: role, jwtFilePath: jwtFilePath, logger: logger}
+}
+
+func (v *VaultClient) GetToken() (string, error) {
+	// read JWT file. JWT is used to authenticate against Vault
```

Comment:
does it makes sense to move this function to fybrik repo under pkg/vault so that other connectors could use it?

Reply:
yes, but I suggest that we postpone this.
There is already a `pkg/vault` library in fybrik, and we need to decide what to do with it.
If we remove it altogether, then maybe we should replace it with this code.